File: android/app/src/main/java/voidawt/GraphicsEnvironment.java
package voidawt;

/** Singleton graphics environment exposing the one {@link GraphicsDevice}. */
public class GraphicsEnvironment {
    private static final GraphicsEnvironment INSTANCE = new GraphicsEnvironment();
    private final GraphicsDevice device = new GraphicsDevice();

    public static GraphicsEnvironment getLocalGraphicsEnvironment() {
        return INSTANCE;
    }

    public GraphicsDevice getDefaultScreenDevice() {
        return device;
    }

    public GraphicsDevice[] getScreenDevices() {
        return new GraphicsDevice[]{device};
    }

    public boolean isHeadlessInstance() {
        return false;
    }

    public static boolean isHeadless() {
        return false;
    }
}

File: src/commonMain/kotlin/awt/GraphicsEnvironment.kt
package awt


expect abstract class GraphicsEnvironment {
//    val screenDevices: Array<GraphicsDevice>
//    val defaultScreenDevice: GraphicsDevice
//    val availableFontFamilyNames: Array<String>
    fun isHeadlessInstance(): Boolean
}

expect fun GraphicsEnvironment.getScreenDevices(): Array<GraphicsDevice>
expect fun GraphicsEnvironment.getDefaultScreenDevice(): GraphicsDevice

expect val localGraphicsEnvironment: GraphicsEnvironment

expect val isHeadless: Boolean
